### PR TITLE
enable pytest-kat/envoy/envoy-v2 in github-actions

### PR DESCRIPTION
**Pull request by anodelman**
- split up kat tests to be able to run on the provided github-actions
  hosts
  * alphabetically (a-h, i-p, q-z)
  * support a new command line option for pytest (--letter-range) to
    indicate which set of tests to execute

Signed-off-by: Alice Nodelman <anode@datawire.io>

## Description
Enable kat/envoy/envoy-v2 tests in github-actions.

## Related Issues
List related issues.

## Testing
Enabled push and executed in github-actions.  Ran green.

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.
   
   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations
 
 - [X] This is unlikely to impact how Ambassador performs at scale.
 
   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability
 
 - [X] My change is adequately tested.
 
   Remember when considering testing:
    + LEGACY MODE TESTS DO NOT RUN FOR EVERY PR. If your change is affected by legacy mode, you need
      to run legacy-mode tests manually (set `AMBASSADOR_LEGACY_MODE=true` and run the tests).
      (This will be fixed soon.)
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points
 
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

File: python/tests/kat/t_mappingtests.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def config(self):
 hostname: "*"
 prefix: /{self.name}/status/
 rewrite: /status/
-service: httpbin.default
+service: httpbin.plain-namespace
```
Comment (kflynn) [inline python/tests/kat/t_mappingtests.py:308]: Huh. Why is this necessary?

File: python/tests/kat/test_ambassador.py [hunk 2]
```diff
@@ -1,47 +1,57 @@
...
+	import t_ip_allow_deny
+	import t_listeneridletimeout
+	import t_loadbalancer
+	import t_logservice
+	import t_lua_scripts
+	import t_mappingtests
+	import t_max_req_header_kb
+	import t_no_ui
+	import t_optiontests
+	import t_plain
```
Comment (kflynn) [inline python/tests/kat/test_ambassador.py:39]: The `plain` thing still worries me. How about comment in here that `plain`, `mappingtests`, and `optiontests` must go together... or, even better, try moving 

```
	import t_mappingtests
	import t_optiontests
```

into `t_plain.py` and let's see what blows up.

**Review by kflynn — CHANGES_REQUESTED** [review]
One major question and one thought about the `plain` thing -- pretty cool though! I didn't know how to add options to `pytest`... 😂

File: python/tests/kat/t_mappingtests.py (hunk 1 above)
Comment (anodelman) [reply python/tests/kat/t_mappingtests.py:308]: There's a question for @acookin

File: python/tests/kat/test_ambassador.py (hunk 2 above)
Comment (anodelman) [reply python/tests/kat/test_ambassador.py]: Will do!

File: python/tests/kat/t_mappingtests.py (hunk 1 above)
Comment (acookin) [reply python/tests/kat/t_mappingtests.py:308]: because that's the naemsapce that the actual service gets deployed to. default was only working because another test was creating it

**Review by acookin — APPROVED** [review]

**Review by acookin — APPROVED** [review]
:shipit:
